packages/@glimmer/reference/test/template-test.ts

```diff
@@ -161,27 +168,28 @@ module('@glimmer/reference: template', () => {
 
     test('it calls getProp', assert => {
       let component = {};
-      let ref = new ComponentRootReference(
-        component,
-        new (class extends TestEnv {
-          getProp(obj: unknown, key: string) {
-            assert.equal(obj, component);
-            assert.equal(key, 'foo');
+      let ref = new ComponentRootReference(component);
 
-            return 123;
-          }
-        })()
-      );
+      let originalContext = testOverrideGlobalContext!({
+        getProp(obj: unknown, key: string) {
+          assert.equal(obj, component);
+          assert.equal(key, 'foo');
+
+          return 123;
+        },
+      });
 
       let fooRef = ref.get('foo');
       assert.equal(fooRef.value(), 123);
+
+      testClearGlobalContextOverride!(originalContext);
```

Comment:
This should be in a try/finally:

```suggestion
      let originalContext = testOverrideGlobalContext!({
        getProp(obj: unknown, key: string) {
          assert.equal(obj, component);
          assert.equal(key, 'foo');

          return 123;
        },
      });

      try {
        let fooRef = ref.get('foo');
        assert.equal(fooRef.value(), 123);
      } finally {
        testClearGlobalContextOverride!(originalContext);
      }
```